Common/src/gRPC/Services/GrpcResultsService.cs

```diff
@@ -125,4 +131,138 @@ public class GrpcResultsService : Results.ResultsBase
              },
            };
   }
+
+  [RequiresPermission(typeof(GrpcResultsService),
+                      nameof(CreateResults))]
+  public override async Task<CreateResultsResponse> CreateResults(CreateResultsRequest request,
+                                                                  ServerCallContext    context)
+  {
+    var results = await request.Results.Select(async rc =>
+                                               {
+                                                 var result = new Result(request.SessionId,
+                                                                         Guid.NewGuid()
+                                                                             .ToString(),
+                                                                         rc.Name,
+                                                                         "",
+                                                                         ResultStatus.Created,
+                                                                         new List<string>(),
+                                                                         DateTime.UtcNow,
+                                                                         Array.Empty<byte>());
+
+                                                 await objectStorage_.AddOrUpdateAsync(result.ResultId,
+                                                                                       new List<ReadOnlyMemory<byte>>
+                                                                                       {
+                                                                                         rc.Data.Memory,
+                                                                                       }.ToAsyncEnumerable(),
+                                                                                       context.CancellationToken)
+                                                                     .ConfigureAwait(false);
+
+                                                 return result;
+                                               })
+                               .WhenAll()
+                               .ConfigureAwait(false);
+
+    var resultList = await results.Select(async r => await resultTable_.CompleteResult(request.SessionId,
```

Comment:
I think you could "merge" this select with the previous one. Besides, you could use ParallelSelect if you want.

Reply:
parallel select was not existing when I wrote the code.

Reply:
well there is an issue with this code. Results are never created in db